answers.txt

```diff
@@ -1,4 +1,34 @@
-� What is a gitignore ?le and what should it include for a Java project? 
+� What is a gitignore file and what should it include for a Java project?
+
+A gitignore file contains types of files to be ignored by git while checking for untracked. This is done to avoid annoying prompts from git about files that are never going to get pushed / commit to the repo.
+For a java project, the gitignore file should include  :
+*.class
+*.log
+*.ctxt
+.mtj.tmp/
+*.jar
+*.war
+*.nar
+*.ear
+*.zip
+*.tar.gz
+*.rar
+hs_err_pid*
+
+
 � What does fast-forward in Git mean? 
-� Why should you have di?erent branches? 
-� In a project how many remote Git repositories do you usually have?
\ No newline at end of file
+
+When we try to merge a commit with another, if the branch on which the merge 
+is being made hasn't diverged or changed, then git, instead of creating a new commit, 
```

Comment:
It does create a new commit, also merge happens but without any issue like merge conflicts.

Reply:
Hi @abhinawsarang , please refer to the link here : [https://sandofsky.com/images/fast_forward.pdf](https://sandofsky.com/images/fast_forward.pdf). It states that a new commit is not created.